### PR TITLE
Use-configurable-pr-number

### PR DESCRIPTION
**Pull request by oredavids**
Pulls in changes from [v0.1.0](https://github.com/gruntwork-io/pipelines-status-update/releases/tag/v0.1.0) and [allows a configurable PR number](https://github.com/gruntwork-io/pipelines-status-update/pull/2/commits/6421c65888aa4ac2fb449379f0cc13ad294dced0).

### CODE REVIEW

File: action.yml [hunk 1]
```diff
@@ -6,26 +6,39 @@ inputs:
     required: true
   step_status:
     description: "The status of the step to update"
-    default: 'not_started'
+    default: "not_started"
+  step_details_preview:
+    description: "Preview Text for .details"
```
Comment (yhakbar) [inline action.yml]: Should this be `details`?

File: action.yml [hunk 2]
```diff
@@ -60,41 +73,56 @@ runs:
         echo "details:"
         cat details.json
         echo "building new json object.."
-        NEWJSON=$(jq '{ "${{ inputs.step_name }}": { "name": "${{ inputs.step_name }}", "status": "${{ inputs.step_status }}", "statusIcon": "'$icon'", "details": . }}' details.json)
+        NEWJSON=$(jq '{ "${{ inputs.step_name }}": { "name": "${{ inputs.step_name }}", "status": "${{ inputs.step_status }}", "statusIcon": "'$icon'", "details_preview": "${{ inputs.step_details_preview }}", "details": . }}' details.json)
```
Comment (yhakbar) [inline action.yml:74]: NIT: We should move these out into `env`.

File: action.yml [hunk 3]
```diff
@@ -60,41 +73,56 @@ runs:
...
         echo "## Gruntwork Pipelines Status" > $TMPFILE
+        echo "<table><tr><th>Step</th><th>Status</th><th>Output</th></tr>" >> $TMPFILE
         IFS=$'\n'
         for item in $(cat state.json | jq -c '. | values[]'); do
             echo "single item..."
             echo $item
             name=$(echo "$item" | jq -r '.name')
             statusIcon=$(echo "$item" | jq -r '.statusIcon')
             details=$(echo "$item" | jq -r '.details')
+            details_preview=$(echo "$item" | jq -r '.details_preview')
```
Comment (yhakbar) [inline action.yml:90]: NIT: You don't need to `echo` here. You can `jq -r '.details_preview' <<< "$item"` I recommend quoting the whole shell substitution as well. i.e. `details_preview="$(...)"`

File: action.yml [hunk 4]
```diff
@@ -60,41 +73,56 @@ runs:
...
         cat updated_state.json
         mv updated_state.json state.json
         echo "## Gruntwork Pipelines Status" > $TMPFILE
+        echo "<table><tr><th>Step</th><th>Status</th><th>Output</th></tr>" >> $TMPFILE
         IFS=$'\n'
         for item in $(cat state.json | jq -c '. | values[]'); do
             echo "single item..."
             echo $item
             name=$(echo "$item" | jq -r '.name')
             statusIcon=$(echo "$item" | jq -r '.statusIcon')
```
Comment (yhakbar) [inline action.yml:88]: NIT: `statusIcon` should be snake_case

File: action.yml (hunk 1 above)
Comment (oredavids) [reply action.yml]: It is the text that shows up next to the details dropdown.

Comment (yhakbar) [reply action.yml]: I mean should the `.` be there?

File: action.yml (hunk 4 above)
Comment (oredavids) [reply action.yml:88]: You may have missed my comment in the PR description; the bulk of the changes are already released in a version. I pulled them in to bring `main` up to date.

Comment (oredavids) [reply action.yml:88]: I can make these changes in a separate PR. Need to use my changes quickly.

**Review by yhakbar — DISMISSED** [review]

**Review by yhakbar — APPROVED** [review]

**Pull request merged**